src/components/SideBar/SideBar.tsx

```diff
@@ -1,29 +1,45 @@
-import React, { useRef } from 'react'
 import {
-  ChevronDoubleLeftIcon,
-  ExclamationIcon,
-  ClockIcon,
-  ExclamationCircleIcon
-} from '@heroicons/react/outline'
+  MutableRefObject,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState
+} from 'react'
+import { ExclamationIcon, ChevronLeftIcon } from '@heroicons/react/outline'
+import { composeClasses } from 'lib/classes'
 import Text from '../Typography'
 import ToolTipHover from '../ToolTipHover'
-import './sideBar.css'
-
-import { composeClasses } from 'lib/classes'
 import Flex from '../Layout/Flex/Flex'
-import Skeleton from '../Skeleton/Skeleton'
+import SideBarHeader from './SideBarHeader'
+import SideBarItem from './SideBarItem'
+import SkeletonSideBarList from './SkeletonSideBarList'
+import './sideBar.css'
 
-export interface SideBarProps {
-  sideBarList?: Array<{
+export interface ISideBarSubItem {
+  [key: string]: {
     title: string
     active: boolean
     to: () => void
-    icon?: JSX.Element
-    disabled?: boolean
-    hidden?: boolean
-  }>
+  }
+}
+
+export type TBadge = string | number | ReactElement
+export interface ISideBarItem {
```

Comment:
```suggestion
export interface SideBarItemProps {
```

Reply:
There is already a SideBarItemProps interface on the SideBarItem componente, ISideBarItem does not represent as such the props that a component receives

Reply:
Ok, but we don't need add suffix to declare an interface, i reffered to this and not for the name exactly